### PR TITLE
[SPARK-13277][BUILD] Follow-up ANTLR warnings are treated as build errors

### PR DESCRIPTION
**Pull request by hvanhovell**
It is possible to create faulty but legal ANTLR grammars. ANTLR will produce warnings but also a valid compileable parser. This PR makes sure we treat such warnings as build errors.

cc @rxin / @viirya

### CODE REVIEW

**Comment by hvanhovell** [comment]
I tested this on my local machine by introducing a warning in ANTLR. It would be great if someone has a suggestion on how we should test this in Jenkins.

**Comment by rxin** [comment]
You can introduce a warning and see if the build fails as part of this pr.

**Comment by hvanhovell** [comment]
Ok I have added the warning. The next merge build should fail.

Build #51131 should pass.

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -434,8 +434,11 @@ object Catalyst {
 
       // Generate the parser.
       antlr.process
```
Comment (rxin) [inline project/SparkBuild.scala]: just a nit: if process is an action, we should add parentheses to it.

**Comment by rxin** [comment]
LGTM if it is creating the proper errors on Jenkins.

**Comment by viirya** [comment]
LGTM, looks like jenkins has generated proposed warning and caught it correctly.

**Comment by rxin** [comment]
Thanks - going to merge this in master.

**Pull request closed without merging**